homeassistant/components/xiaomi_miio/select.py

```diff
@@ -0,0 +1,253 @@
+        coordinator = hass.data[DOMAIN][config_entry.entry_id][KEY_COORDINATOR]
+        device_features = 0
+        entity_class = None
+
+        _LOGGER.debug("Initializing with host %s (token %s...)", host, token[:5])
+
+        if model in [MODEL_AIRHUMIDIFIER_CA1, MODEL_AIRHUMIDIFIER_CB1]:
+            device_features = FEATURE_FLAGS_AIRHUMIDIFIER_CA_AND_CB
+            entity_class = XiaomiAirHumidifierSelector
+        elif model in [MODEL_AIRHUMIDIFIER_CA4]:
+            device_features = FEATURE_FLAGS_AIRHUMIDIFIER_CA4
+            entity_class = XiaomiAirHumidifierMiotSelector
+        elif model in MODELS_HUMIDIFIER:
+            device_features = FEATURE_FLAGS_AIRHUMIDIFIER
+            entity_class = XiaomiAirHumidifierSelector
+        else:
+            return
+
+        for feature in SELECTOR_TYPES:
+            selector = SELECTOR_TYPES[feature]
+            if feature & device_features and feature in SELECTOR_TYPES:
+                entities.append(
+                    entity_class(
+                        f"{config_entry.title} {selector.name}",
+                        device,
+                        config_entry,
+                        f"{selector.short_name}_{config_entry.unique_id}",
+                        selector,
+                        coordinator,
+                    )
+                )
+
+    async_add_entities(entities, update_before_add=False)
+
+
+class XiaomiSelector(XiaomiCoordinatedMiioEntity, SelectEntity):
+    """Representation of a generic Xiaomi attribute selector."""
+
+    def __init__(self, name, device, entry, unique_id, selector, coordinator):
+        """Initialize the generic Xiaomi attribute selector."""
+        super().__init__(name, device, entry, unique_id, coordinator)
+        self._state = None
+        self._attr_icon = selector.icon
+        self._attr_unit_of_measurement = selector.unit_of_measurement
+        self._supported_features = 0
+        self._device_features = 0
+        self._state_attrs = {}
+        self._controller = selector
+        self._current_option = None
+        self._enum_class = None
+        self._attributes = None
+        self._available_attributes = []
+
+    @property
+    def options(self):
+        """Return available options."""
+        return self._controller.options
+
+    @property
+    def available(self):
+        """Return true when state is known."""
+        return super().available and self._available
+
+    @property
+    def current_option(self):
+        """Return the current option."""
+        if not self.available and self.coordinator.data:
+            self._available = True
+            self._current_option = self._extract_value_from_attribute(
+                self.coordinator.data, self._controller.short_name
+            )
+        return (
+            getattr(self, SERVICE_TO_METHOD[self._controller.service]["property"])
+            if self.available
+            else None
+        )
```

Comment:
This is not necessary.
```suggestion
        return getattr(self, SERVICE_TO_METHOD[self._controller.service]["property"])
```

Reply:
Do we not need to check for self.available here?